pkg/security/probe/probe.go

```diff
@@ -197,15 +249,20 @@ func (p *Probe) AddCustomEventHandler(eventType model.EventType, handler CustomE
 	return nil
 }
 
-func (p *Probe) sendEventToWildcardHandlers(event *model.Event) {
-	for _, handler := range p.fullAccessEventHandlers[model.UnknownEventType] {
+func (p *Probe) sendEventToHandlers(event *model.Event) {
+	for _, handler := range p.eventHandlers[model.UnknownEventType] {
 		handler.HandleEvent(event)
 	}
 }
 
-func (p *Probe) sendEventToSpecificEventTypeHandlers(event *model.Event) {
-	for _, handler := range p.eventHandlers[event.GetEventType()] {
-		handler.HandleEvent(handler.Copy(event))
+func (p *Probe) sendEventToConsumers(event *model.Event) {
+	for _, pc := range p.eventConsumers[event.GetEventType()] {
+		if copied := pc.consumer.Copy(event); copied != nil {
```

Comment:
What's the impact of a "slow" copy during load?
The copy is implemented by the "consumer", so we have no guard for implementing a safe and fast copy, which can affect other consumers

Reply:
It depends what the consumer are going to copy and how. I decided to put no guard in the copy process to avoid thing that would cause performance impact. I think we can put unit test, CI jobs to enforce the "safe" and "fast" copy.

Reply:
I added a generator which convert struct to the copy function. Of course someone can create its own copy function but I think this is already a good first step to limit the things that the copy function can do.